packages/block-editor/src/store/reducer.js

```diff
@@ -192,6 +195,154 @@ export function isUpdatingSameBlockAttribute( action, lastAction ) {
 	);
 }
 
+/**
+ * Higher-order reducer intended to reset the cache key of all blocks
+ * whenever the post meta values change.
+ *
+ * @param {Function} reducer Original reducer function.
+ *
+ * @return {Function} Enhanced reducer function.
+ */
+const withPostMetaUpdateCacheReset = ( reducer ) => ( state, action ) => {
+	const newState = reducer( state, action );
+	const previousMetaValues = get( state.settings.__experimentalMetaSource, [ 'value' ] );
+	const nextMetaValues = get( newState.settings.__experimentalMetaSource, [ 'value' ] );
+	// If post meta values change, reset the cache key for all blocks
+	if ( previousMetaValues !== nextMetaValues ) {
+		newState.blocks = {
```

Comment:
Something to look into separately: This condition is satisfied when saving a post, regardless of whether there are any updates to post meta. Possibly redundant with custom sources work, depending on whether there's an issue with how we're updating/resetting post during save and needlessly creating/assigning new meta references when no effective changes have happened.